File: src/main/java/com/vaadin/demo/component/tabs/TabsStates.java
package com.vaadin.demo.component.tabs;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.tabs.Tab;
import com.vaadin.flow.component.tabs.Tabs;
import com.vaadin.flow.router.Route;

@Route("tabs-states")
public class TabsStates extends Div {

	public TabsStates() {
		// tag::snippet[]
		Tab selected = new Tab("Selected");

		Tab unselected = new Tab("Unselected");

		Tab disabled = new Tab("Disabled");
		disabled.setEnabled(false);

		Tabs tabs = new Tabs(selected, unselected, disabled);
		tabs.setSelectedTab(selected);
		// end::snippet[]
		add(tabs);
	}

	public static class Exporter extends DemoExporter<TabsStates> { // hidden-source-line
	} // hidden-source-line
}

File: src/main/frontend/demo/component/tabs/tabs-states.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/tabs';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';

@customElement('tabs-states')
export class Example extends LitElement {
  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-tabs>
        <vaadin-tab>Selected</vaadin-tab>
        <vaadin-tab>Unselected</vaadin-tab>
        <vaadin-tab disabled>Disabled</vaadin-tab>
      </vaadin-tabs>
      <!-- end::snippet[] -->
    `;
  }
}
